src/index.ts

```diff
@@ -0,0 +1,158 @@
+// eslint-disable-next-line @typescript-eslint/ban-ts-ignore
+// @ts-ignore
+const { getPromiseDetails } = process.binding('util');
+
+/**
+ * The class for deep checking Types
+ */
+export default class Type {
```

Comment:
Unrelated but, this should be
```suggestion
export class Type {
```

Because likewise in all the utils, we will import the utils by destructuring them, only one or two utils export a single thing, so you know.